File: tests/msbuild/vbnet/src/nBuildKit.Test.VbNet.Library/HelloWorld.vb
Imports System.Globalization
Imports System.Reflection

Public Class HelloWorld
    Private ReadOnly _name As String = AssemblyName()

    Private ReadOnly _version As String = AssemblyVersion()

    Public Function SayHello() As String
        Return String.Format( _
            CultureInfo.InvariantCulture, _
            "Hello world from: {0} [{1}]", _
            _name, _
            _version)
    End Function

    Private Function AssemblyName() As String
        Dim attr As AssemblyTitleAttribute = TryCast(Assembly.GetExecutingAssembly().GetCustomAttribute(GetType(AssemblyTitleAttribute)), AssemblyTitleAttribute)
        Return attr.Title
    End Function

    Private Function AssemblyVersion() As String
        Dim attr As AssemblyInformationalVersionAttribute = TryCast(Assembly.GetExecutingAssembly().GetCustomAttribute(GetType(AssemblyInformationalVersionAttribute)), AssemblyInformationalVersionAttribute)
        Return attr.InformationalVersion
    End Function
End Class

File: tests/latest/csharp/src/nBuildKit.Test.CSharp.Library/HelloWorld.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace NBuildKit.Test.CSharp.Library
{
    /// <summary>
    /// Provides methods to say hello to the world.
    /// </summary>
    public class HelloWorld
    {
        private readonly string _name = AssemblyName();
        private readonly string _version = AssemblyVersion();

        /// <summary>
        /// Says hello to the world.
        /// </summary>
        /// <returns>A string containing the message.</returns>
        public string SayHello()
        {
            return string.Format(
                CultureInfo.InvariantCulture,
                "Hello world from: {0} [{1}]",
                _name,
                _version);
        }

        private static string AssemblyName()
        {
            var attribute = (AssemblyTitleAttribute)Assembly.GetExecutingAssembly().GetCustomAttribute(typeof(AssemblyTitleAttribute));
            return attribute.Title;
        }

        private static string AssemblyVersion()
        {
            var attribute = (AssemblyInformationalVersionAttribute)Assembly.GetExecutingAssembly().GetCustomAttribute(typeof(AssemblyInformationalVersionAttribute));
            return attribute.InformationalVersion;
        }
    }
}
